### PR TITLE
[pkg/oci]: Add Delete function

### PR DESCRIPTION
**Pull request by somtochiama**
Part of: [fluxv2/#2983](https://github.com/fluxcd/flux2/issues/2982)

### CODE REVIEW

**Comment by somtochiama** [comment]
Failing git-test is fixed #318

File: oci/client/internal/fs/fs_test.go [hunk 1]
```diff
@@ -41,7 +41,6 @@ func TestMain(m *testing.M) {
 	}
 
 	os.MkdirAll("testdata/symlinks", 0o755)
-	defer os.RemoveAll("testdata/symlinks")
```
Comment (stefanprodan) [inline oci/client/internal/fs/fs_test.go:44]: Are OCI tests broken in main? If so please move this to a different PR as it's not related to client.delete.

Comment (somtochiama) [reply oci/client/internal/fs/fs_test.go:44]: They aren't broken per se. The defer function never runs because of os.Exit() so the created directory doesn't get removed.

But you are right. I would move it to a different pr

**Comment by stefanprodan** [comment]
@somtochiama please rebase and mark the PR ready for review.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @somtochiama
